snapcraft/internal/repo/snaps.py

```diff
@@ -270,14 +275,25 @@ def download_snaps(*, snaps_list: Sequence[str], directory: str) -> None:
         snap_pkg.download(directory=directory)
 
 
-def install_snaps(snaps_list):
+def install_snaps(snaps_list: Union[Sequence[str], Set[str]]) -> List[str]:
     """Install snaps of the format <snap-name>/<channel>.
 
     :return: a list of "name=revision" for the snaps installed.
     """
     snaps_installed = []
     for snap in snaps_list:
         snap_pkg = SnapPackage(snap)
+
+        # Allow bases to be installed from non stable channels.
+        snap_pkg_channel = snap_pkg.get_store_snap_info()["channel"]
```

Comment:
Does this work if the snap is not installed?  Which "channel" would be the default provided? stable, if available?

Reply:
This is information from the store, not local

Reply:
Here's an illustration for various snaps and the last one showing the usual workflow when adding an invalid snap:
```
In [2]: from snapcraft.internal.repo import snaps                                                          

In [3]: snaps.SnapPackage("lxd").get_store_snap_info()["channel"]                                          
Out[3]: 'stable'

In [4]: snaps.SnapPackage("core20").get_store_snap_info()["channel"]                                       
Out[4]: 'edge'

In [5]: snaps.SnapPackage("core18").get_store_snap_info()["channel"]                                       
Out[5]: 'stable'

In [6]: snaps.SnapPackage("snapcraft").get_store_snap_info()["channel"]                                    
Out[6]: 'stable'

In [7]: snaps.SnapPackage("foo").get_store_snap_info()["channel"]                                          
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
~/snapcraft/snapcraft/internal/repo/snaps.py in get_store_snap_info(self)
    113                 try:
--> 114                     self._store_snap_info = _get_store_snap_info(self.name)
    115                     break

~/snapcraft/snapcraft/internal/repo/snaps.py in _get_store_snap_info(snap_name)
    385     snap_info = requests_unixsocket.get(url)
--> 386     snap_info.raise_for_status()
    387     return snap_info.json()["result"][0]

~/.venv/snapcraft/lib/python3.6/site-packages/requests/models.py in raise_for_status(self)
    939         if http_error_msg:
--> 940             raise HTTPError(http_error_msg, response=self)
    941 

HTTPError: 404 Client Error: Not Found for url: http+unix://%2Frun%2Fsnapd.socket/v2/find?name=foo

During handling of the above exception, another exception occurred:

SnapUnavailableError                      Traceback (most recent call last)
<ipython-input-7-b9705a1c80c5> in <module>
----> 1 snaps.SnapPackage("foo").get_store_snap_info()["channel"]

~/snapcraft/snapcraft/internal/repo/snaps.py in get_store_snap_info(self)
    123                     if http_error.response.status_code == 404:
    124                         raise errors.SnapUnavailableError(
--> 125                             snap_name=self.name, snap_channel=self.channel
    126                         )
    127                     retry_count -= 1

SnapUnavailableError: Failed to install or refresh a snap: 'foo' does not exist or is not available on the desired channel 'latest/stable'. Use `snap info foo` to get a list of channels the snap is available on.
```